### PR TITLE
is1047/remove-minio-and-webclient

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- [x] integrate new qooxdoo-kit into workflow (see ``services/web/client/README.md``)
   - cleanup of web/client folder and new doc
   - remove web_client service
- [x] simplified main Makefile by splitting build options

## Related issue number
implementing issue #1047 

## How to test
Basically it should work as before 

     $ make build-devel
     $ make up-devel  # (now this is IDENTICAL to make up-webclient-devel
check website, modify webclient code, ... refresh

     $ make build up-prod

**NOTICE**: ``up-``**prod**

@odeimaiz  please test if everything works fine in windows
@odeimaiz, @ignapas  please make sure you are comfortable in dev mode for the front-end (please see [README](services/web/client/README.md) and [Makefile](services/web/client/Makefile) for more details and alternatives

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

**Comment by pcrespov** [comment]
@odeimaiz CI fails in ops/travis/unit-testing/frontend because the ``package.json`` is not there anymore. I rather discuss with you how to modify this.

File: Makefile [hunk 1]
```diff
@@ -62,54 +62,66 @@ TEMP_COMPOSE_YML := $(shell $(if $(IS_WIN), (New-TemporaryFile).FullName, mktemp
...
+build: .env ## Builds all core service images (user 'make build-nc' to build w/o cache)
+	# Compiling front-end
+	$(MAKE) -C services/web/client compile
+	# Building services
+	export BUILD_TARGET=production; \
+	$(DOCKER_COMPOSE) -f services/docker-compose.build.yml build --parallel
+
+
+.PHONY: rebuild build-nc
+rebuild: build-nc #TODO: deprecate rebuild
```
Comment (sanderegg) [inline Makefile]: nc? well.. not sure that is clearer than rebuild;)

File: Makefile [hunk 2]
```diff
@@ -62,54 +62,66 @@ TEMP_COMPOSE_YML := $(shell $(if $(IS_WIN), (New-TemporaryFile).FullName, mktemp
...
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml build --no-cache --parallel $(SERVICES_LIST)
 
 .PHONY: build-devel
-build-devel: .env .build-webclient ## Builds images of core services for development
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.devel.yml build --parallel
+build-devel: .env ## Builds images of core services for development
+	# Compiling front-end ('compile' target needed since productions stage in Dockerfile of webserver copies client/tools/qooxdoo-kit/build:latest)
+	$(MAKE) -C services/web/client compile compile-dev
+	# Building services
+	export BUILD_TARGET=development; \
```
Comment (sanderegg) [inline Makefile]: this is a bit weird, so even though the docker-compose.devel.yml is used, one needs to export the BUILD_TARGET?

File: Makefile [hunk 3]
```diff
@@ -129,19 +141,18 @@ down-force: ## forces to stop all services and leave swarms
...
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.cache.yml pull
+	$(DOCKER_COMPOSE) -f services/docker-compose.cache.yml pull
 
 .PHONY: build-cache
 build-cache: ## Builds service images and tags them as 'cache'
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.cache.yml build --parallel $(filter-out webserver, $(CACHED_SERVICES_LIST))
-	$(DOCKER) tag $(DOCKER_REGISTRY)/webclient:cache services_webclient:build
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.cache.yml build webserver
+	$(DOCKER_COMPOSE) -f services/docker-compose.build.yml -f services/docker-compose.cache.yml build --parallel
+	## $(filter-out webserver, $(CACHED_SERVICES_LIST))
```
Comment (sanderegg) [inline Makefile]: line to be removed?

File: Makefile [hunk 4]
```diff
@@ -39,7 +39,7 @@ SERVICES_LIST := \
 	storage \
 	webserver
 
-CACHED_SERVICES_LIST    := $(SERVICES_LIST) webclient
+CACHED_SERVICES_LIST    := $(SERVICES_LIST)
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply Makefile]: > Wow. Much cleaner now. One thing, I do not see anything about removing minio. Is this still coming?

yes. despite the name of the branch, this only includes webclient ... I noticed it was too much for a single PR

**Comment by ignapas** [comment]
> image
> Getting an insta-error when `make up-devel`. Running on Mac.

This is happening to me as well in other branches... have you had this problem before?

File: Makefile (hunk 1 above)
Comment (pcrespov) [reply Makefile]: You have a good eye :-)
I did it for two reasons:
a) help is now alphabetical and shows "build-nc" as a variation of "build".
b) was following the [Makefile XOps blog](https://itnext.io/docker-makefile-x-ops-sharing-infra-as-code-parts-ea6fa0d22946) and he uses there

I will leave ``rebuild`` as an alias. Would that be fine :-)?

File: Makefile (hunk 2 above)
Comment (pcrespov) [reply Makefile]: yes I explain that to you better in person .. it is easier, provided the extra changes i have added

**Comment by pcrespov** [comment]
> > image
> > Getting an insta-error when `make up-devel`. Running on Mac.
> 
> This is happening to me as well in other branches... have you had this problem before?

Yeah ... this is a problem of compatibility with the mac and ``mktemp`` ... will PR a fast fix for that

File: Makefile [hunk 5]
```diff
@@ -217,15 +282,19 @@ info: ## displays selected parameters of makefile environments
 	@echo '+ DOCKER_REGISTRY      : $(DOCKER_REGISTRY)'
 	@echo '+ DOCKER_IMAGE_TAG     : ${DOCKER_IMAGE_TAG}'
 
-
-.PHONY: info-more
-info-more: ## displays all parameters of makefile environments
+info-vars: ## displays all parameters of makefile environments
```
Comment (mguidon) [inline Makefile]: infowars?

**Comment by ignapas** [comment]
resolve conflict?

**Comment by pcrespov** [comment]
> resolve conflict?

I am working on that :-)

File: Makefile [hunk 6]
```diff
@@ -38,85 +38,126 @@ SERVICES_LIST := \
...
+	$(MAKE) -C services/web/client compile-dev flags=--watch
 
 
 .PHONY: down down-force
-down: ## stops and removes stack
+down: ## Stops and removes stack
 	$(DOCKER) stack rm $(SWARM_STACK_NAME)
 
-down-force: ## forces to stop all services and leave swarms
+down-force: ## Forces to stop all services and leave swarms
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply Makefile]: we stay with ``leave``

File: Makefile [hunk 7]
```diff
@@ -38,85 +38,126 @@ SERVICES_LIST := \
...
+	@$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.prod.yml config > $(output_file)
 
 
-## DOCKER SWARM -------------------------------
 .PHONY: up up-devel
-up: .env .init-swarm ## init swarm and deploys all core and tool services up [-devel suffix uses container in development mode]
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose-tools.yml config > $(TEMP_COMPOSE_YML);
+up: .env .init-swarm ## Deploys production stack (analogous to other osparc-ops's stack)
+	# config stack to $(TEMP_COMPOSE_YML) with '$(DOCKER_REGISTRY)/{service}:$(DOCKER_IMAGE_TAG)'
+	@$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.prod.yml config > $(TEMP_COMPOSE_YML)
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply Makefile]: removed ``up`` and added ``up-prod``, ``up-version`` and ``up-latest`` as agreed

File: Makefile [hunk 8]
```diff
@@ -38,85 +38,141 @@ SERVICES_LIST := \
...
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml build --no-cache --parallel $(SERVICES_LIST)
 
 .PHONY: build-devel
-build-devel: .env .build-webclient ## Builds images of core services for development
-	$(DOCKER_COMPOSE) -f services/docker-compose.yml -f services/docker-compose.devel.yml build --parallel
+build-devel: .env ## Builds development images and tags them as 'local/{service-name}:development'
+	# Compiling front-end ('compile' target needed since productions stage in Dockerfile of webserver copies client/tools/qooxdoo-kit/build:latest)
+	$(MAKE) -C services/web/client compile compile-dev
+	# Building services
+	export BUILD_TARGET=development; \
```
Comment (odeimaiz) [inline Makefile]: Cmder doesn't like the "export" bit

**Comment by ignapas** [comment]
> After executing every resetting command I know:
> 
> ```
> make down
> make clean
> docker system prune -a
> docker image prune -a
> make build-devel
> make up-devel
> ```
> 
> localhost:9081 is not accesible
> Screenshot from 2019-09-25 11-03-06
> Screenshot from 2019-09-25 11-03-40

same thing happens on Mac 
Screenshot from 2019-09-25 11-08-24
Web server won't start

File: Makefile (hunk 8 above)
Comment (pcrespov) [reply Makefile]: @odeimaiz follow [README.txt](README.txt) section on windows and it should work.

**Pull request merged**